# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webaudio/blink/HRTFKernel.cpp
# Commit: 1d36088485a4
# Full Hash: 1d36088485a4ad3034d6a58df21430c3d3b54cc3
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-27 04:32:10
# Regressor Bug: 1879873
# File Overlap Count: 1
# Description:
#   Bug 1879873 - Scale the inverse FFT result using tx instead of pre- or post-processing the data. r=karlt
#   
#   This skips a copy at the expense of having to do a `const_cast`.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202434
# ==============================================================================

diff -r 13edc1b203a7 -r 1d36088485a4 dom/media/webaudio/blink/HRTFKernel.cpp
--- a/dom/media/webaudio/blink/HRTFKernel.cpp	Mon Feb 26 16:07:09 2024 +0000
+++ b/dom/media/webaudio/blink/HRTFKernel.cpp	Mon Feb 26 16:07:09 2024 +0000
@@ -38,7 +38,7 @@
   // Check for power-of-2.
   MOZ_ASSERT(length && (length & (length - 1)) == 0);
 
-  FFTBlock estimationFrame(length);
+  FFTBlock estimationFrame(length, 1.f / length);
   estimationFrame.PerformFFT(impulseP);
 
   float frameDelay =